basic-search-java/src/main/java/com/mydomain/example/MetalSearcher.java

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.mydomain.example;
+
+import com.google.inject.Inject;
+import com.yahoo.prelude.query.IndexedItem;
+import com.yahoo.prelude.query.Item;
+import com.yahoo.prelude.query.OrItem;
+import com.yahoo.prelude.query.WordItem;
+import com.yahoo.search.Query;
+import com.yahoo.search.Result;
+import com.yahoo.search.Searcher;
+import com.yahoo.search.query.Model;
+import com.yahoo.search.query.QueryTree;
+import com.yahoo.search.searchchain.Execution;
+import com.yahoo.yolean.chain.After;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Searcher which adds an OR-term to queries with Metal intent - demonstrates:
+ * <ol>
+ *     <li>How to get the query tree, and modify it</li>
+ *     <li>Use of tracing</li>
+ * </ol>
+ */
+public class MetalSearcher extends Searcher {
+
+    private final List<String> metalWords;
+
+    @Inject
+    public MetalSearcher(MetalNamesConfig config) {
+        metalWords = config.metalWords();
+    }
+
+    public MetalSearcher() {
+        metalWords = new ArrayList<>();
+    }
+
+    @Override
+    public Result search(Query query, Execution execution) {
+        Model model = query.getModel();
+        QueryTree tree = model.getQueryTree();
+        if (!tree.isEmpty()) {
```

Comment:
i suggest you move this condition into the isMetalQuery item so this gets easier to read.

Reply:
Then you could also invert it and do if (tree.isEmpty()) return false;

Reply:
thanks. the searcher was also wrong, fixed